packages/typescript/core/src/core/api.ts

```diff
@@ -259,3 +259,167 @@ export function getLastActiveTraceId(): string | undefined {
+  /**
+   * A dictionary of tags to update the trace with. Tags are designed for mutable values
+   * that can be updated after the trace is created via MLflow UI or API.
+   */
+  tags?: Record<string, string>;
+
+  /**
+   * A dictionary of metadata to update the trace with. Metadata cannot be updated
+   * once the trace is logged. It is suitable for recording immutable values like the
+   * git hash of the application version that produced the trace.
+   */
+  metadata?: Record<string, string>;
+
+  /**
+   * Client supplied request ID to associate with the trace. This is useful for linking
+   * the trace back to a specific request in your application or external system.
+   */
+  clientRequestId?: string;
+
+  /**
+   * A preview of the request to be shown in the Trace list view in the UI.
+   * By default, MLflow will truncate the trace request naively by limiting the length.
+   * This parameter allows you to specify a custom preview string.
+   */
+  requestPreview?: string;
+
+  /**
+   * A preview of the response to be shown in the Trace list view in the UI.
+   * By default, MLflow will truncate the trace response naively by limiting the length.
+   * This parameter allows you to specify a custom preview string.
+   */
+  responsePreview?: string;
+}
+
+/**
+ * Update the current active trace with the given options.
+ *
+ * You can use this function either within a function decorated with `@trace` or
+ * within the scope of the `withSpan` context. If there is no active trace found,
+ * this function will log a warning and return.
+ *
+ * @param options Options for updating the trace
+ *
+ * @example
+ * Using within a function decorated with `@trace`:
+ * ```typescript
+ * const myFunc = trace((x: number) => {
+ *   updateCurrentTrace({ tags: { fruit: "apple" }, clientRequestId: "req-12345" });
+ *   return x + 1;
+ * });
+ * ```
+ *
+ * @example
+ * Using within the `withSpan` context:
+ * ```typescript
+ * withSpan((span) => {
+ *   updateCurrentTrace({ tags: { fruit: "apple" }, clientRequestId: "req-12345" });
+ * }, { name: "span" });
+ * ```
+ *
+ * @example
+ * Updating source information of the trace:
+ * ```typescript
+ * updateCurrentTrace({
+ *   metadata: {
+ *     "mlflow.trace.session": "session-4f855da00427",
+ *     "mlflow.trace.user": "user-id-cc156f29bcfb",
+ *     "mlflow.source.name": "inference.ts",
+ *     "mlflow.source.git.commit": "1234567890",
+ *     "mlflow.source.git.repoURL": "https://github.com/mlflow/mlflow"
+ *   }
+ * });
+ * ```
+ */
+export function updateCurrentTrace(options: UpdateCurrentTraceOptions): void {
+  const activeSpan = getCurrentActiveSpan();
+
+  if (!activeSpan) {
+    console.warn(
+      'No active trace found. Please create a span using `withSpan` or ' +
+        '`@trace` before calling `updateCurrentTrace`.'
+    );
+    return;
+  }
+
+  // Validate string parameters
+  if (options.requestPreview !== undefined && typeof options.requestPreview !== 'string') {
```

Comment:
Do we need to check `options.requestPreview !== undefined` if we check `typeof options.requestPreview !== 'string'`?